e2e/cypress/integration/accessibility/accessibility_buttons_spec.js

```diff
@@ -0,0 +1,80 @@
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @accessibility
+
+describe('Verify Accessibility Support in different Buttons', () => {
+    before(() => {
+        cy.apiLogin('sysadmin');
+
+        // # Visit the Off Topic channel
+        cy.visit('/ad-1/channels/off-topic');
+        cy.findAllByTestId('postView').should('be.visible');
+
+        // # Remove from Favorites if already set
+        cy.get('#channelHeaderInfo').then((el) => {
+            if (el.find('#toggleFavorite.active').length) {
+                cy.get('#toggleFavorite').click();
+            }
+        });
+
+        // # Post a message
+        cy.postMessage(`hello from sysadmin: ${Date.now()}`);
+    });
+
+    it('MM-22624 Accessibility Support in RHS expand and close icons', () => {
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostCommentIcon(postId);
+            cy.get('#rhsContainer').should('be.visible').within(() => {
+                // * Verify accessibility support in Sidebar Expand and Shrink icon
+                cy.get('button.sidebar--right__expand').should('have.attr', 'aria-label', 'Expand').within(() => {
+                    cy.get('.fa-expand').should('have.attr', 'aria-label', 'Expand the sidebar icon');
+                    cy.get('.fa-compress').should('have.attr', 'aria-label', 'Shrink the sidebar icon');
+                });
+
+                // * Verify accessibility support in Close icon
+                cy.get('#rhsCloseButton').should('have.attr', 'aria-label', 'Close').within(() => {
+                    cy.get('i').should('have.attr', 'aria-label', 'Close the sidebar icon');
+                });
+
+                // # Close the sidebar
+                cy.get('#rhsCloseButton').click();
+            });
+        });
+    });
+
+    it('MM-22624 Accessibility Support in different buttons in Channel Header', () => {
+        // # Ensure the focus is on the Toggle Favorites button
+        cy.get('#toggleFavorite').focus().tab({shift: true}).tab();
+
+        // * Verify accessibility support in Favorites button
+        cy.get('#toggleFavorite').should('have.attr', 'aria-label', 'add to favorites').and('have.class', 'a11y--active a11y--focused').click();
+
+        // * Verify accessibility support if Channel is added to Favorites
+        cy.get('#toggleFavorite').should('have.attr', 'aria-label', 'remove from favorites');
+
+        // # Ensure the focus is on the Toggle Favorites button
+        cy.get('#toggleFavorite').focus().tab({shift: true}).tab();
+
+        // # Press tab until the focus is on the Pinned posts button
+        cy.focused().tab().tab().tab().tab();
```

Comment:
This generally works fine, but in my case where I have a plugin that has an icon in the header between the Pin and Members icons, it's failing. Not sure if we want to tackle this scenario right now to check if the focus is on the pin button?

Reply:
@prapti It will be difficult to tackle this, i.e. there might be some other plugins locally installed which the test might not know. It will be too much to set up a plugin in this test as its not dependent & out of scope of this test & generally on the Cypress test instance we would not have the plugin enabled while running this test. Hence I think it would be okay to skip this for now. 
<img width="1564" alt="Screenshot 2020-04-14 at 6 10 54 PM" src="https://user-images.githubusercontent.com/1429138/79226156-87a47280-7e7b-11ea-9e02-3cb99a3bef1c.png">